src/protagonist/DLCS.AWS/SNS/TopicPublisher.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Amazon.SimpleNotificationService;
+using Amazon.SimpleNotificationService.Model;
+using DLCS.AWS.Settings;
+using DLCS.Core;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace DLCS.AWS.SNS;
+
+public class TopicPublisher : ITopicPublisher, IDisposable
+{
+    private IAmazonSimpleNotificationService client;
+    private ILogger<TopicPublisher> logger;
+    private SNSSettings sNSSettings;
+
+    public TopicPublisher(IAmazonSimpleNotificationService client, ILogger<TopicPublisher> logger, IOptions<AWSSettings> settings)
+    {
+        this.client = client;
+        this.logger = logger;
+        sNSSettings = settings.Value.SNS;
+    }
+    
+    /// <inheritdoc />
+    public async Task<bool> PublishToTopicAsync(string messageContents, SubscribedQueueType subscribedQueueType, CancellationToken cancellationToken = default)
+    {
+        CreateTopicResponse? topic; 
+        try
+        {
+            // Retrieving the ARN of a topic from AWS requires calling CreateTopic
+            topic = await client.CreateTopicAsync(sNSSettings.DeleteNotificationTopicName, cancellationToken);
```

Comment:
This feels odd. Would this mean that the API would require permission for `sns:CreateTopic` action to raise a notification? Storing the ARN of the topic would be simpler?

Reply:
to be honest, I'd need to check what permissions it needs exactly! Think I was probably being too clever here by trying to pull it from the name - given the ARN isn't going to change, it's probably less confusing if I just use the ARN directly